daft/io/parquet.py

```diff
@@ -43,16 +45,34 @@ def read_parquet(
     returns:
         DataFrame: parsed DataFrame
     """
+
     if isinstance(path, list) and len(path) == 0:
         raise ValueError(f"Cannot read DataFrame from from empty list of Parquet filepaths")
 
-    plan = _get_tabular_files_scan(
-        path,
-        schema_hints,
-        ParquetSourceInfo(
-            io_config=io_config,
-            use_native_downloader=use_native_downloader,
-        ),
-        fs,
-    )
+    context = get_context()
+
+    if context.use_rust_planner:
```

Comment:
Some quick questions about the current "switching between all-Python logical plan vs. Rust-based logical plan builder" setup.

We previously talked about exposing a `LogicalPlanBuilder` interface that would have two implementations, the all-Python query planner (`PyLogicalPlanBuilder`) and the new Rust query planner (`RustLogicalPlanBuilder`), where the `DataFrame` API layer would be rewritten to use that `LogicalPlanBuilder` interface and we could limit the number of places we'd need to switch on `context.use_rust_planner`. E.g. the `DataFrame` constructor would take a `LogicalPlanBuilder` instance instead of a union of the all-Python `LogicalPlan` and the `RustLogicalPlanBuilder`:
```python
    def __init__(self, builder: LogicalPlanBuilder) -> None:
        if not isinstance(plan, LogicalPlanBuilder):
            if isinstance(plan, dict):
                raise ValueError(
                    f"DataFrames should be constructed with a dictionary of columns using `daft.from_pydict`"
                )
            if isinstance(plan, list):
                raise ValueError(
                    f"DataFrames should be constructed with a list of dictionaries using `daft.from_pylist`"
                )
            raise ValueError(f"Expected DataFrame to be constructed with a LogicalPlan, received: {plan}")

        self.__plan = plan
        self._result_cache: Optional[PartitionCacheEntry] = None
        self._preview = DataFramePreview(preview_partition=None, dataframe_num_rows=None)
```

And `read_parquet()` would look something like this:
```python
def read_parquet(
    path: Union[str, List[str]],
    schema_hints: Optional[Dict[str, DataType]] = None,
    fs: Optional[fsspec.AbstractFileSystem] = None,
    io_config: Optional["IOConfig"] = None,
    use_native_downloader: bool = False,
) -> DataFrame:
    if isinstance(path, list) and len(path) == 0:
        raise ValueError(f"Cannot read DataFrame from from empty list of Parquet filepaths")

    context = get_context()
    builder= RustLogicalPlanBuilder() if context.use_rust_planner else PyLogicalPlanBuilder()
    new_builder = builder.scan(
        path,
        schema_hints,
        ParquetSourceInfo(
            io_config=io_config,
            use_native_downloader=use_native_downloader,
        ),
        fs,
    )
    return DataFrame(new_builder)
```

This should end up being a good bit cleaner than imperatively switching between implementations within each `DataFrame` API function/method, and we _should_ be able to line up the builder interfaces (such as `builder.scan()` or `builder.filter()`) since they're adding the same logical op to each underlying logical plan. This does, however, require more upfront changes to the `DataFrame` implementation, since each method would need to be ported to the `PyLogicalPlanBuilder` interface (although this should be straightforward).

Do you agree that this is a better long-term approach, and if so, are you thinking that deferring this refactor is the best choice for now?